internal/gencli/flag.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gencli
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/googleapis/gapic-generator-go/internal/pbinfo"
+)
+
+// Flag is used to represent fields as flags
+type Flag struct {
+	Name          string
+	Type          descriptor.FieldDescriptorProto_Type
+	Message       string
+	Repeated      bool
+	Required      bool
+	Usage         string
+	MessageImport pbinfo.ImportSpec
+	OneOfs        map[string]*Flag
+	IsOneOfField  bool
+	IsNested      bool
+}
+
+// GenOtherVarName generates the Go variable to store repeated Message & Enum string values
+func (f *Flag) GenOtherVarName(in string) string {
+	return in + strings.Replace(f.InputFieldName(), ".", "", -1)
+}
+
+// GenOneOfVarName generates the variable name for a oneof entry type
+func (f *Flag) GenOneOfVarName(in string) string {
+	name := f.InputFieldName()
+	if !f.IsNested && strings.Count(f.Name, ".") > 1 {
```

Comment:
```
if p := strings.LastIndex(f.Name, "."); p >= 0 && !f.IsNested {
  name = f.Name[:p]
}
```

Reply:
I cannot do this because the fields of nested messages are named using dot-notation i.e. `--size.width` and `oneof` fields I prefix with the `oneof`'s name, followed by the field name. Thus, a `oneof` field that is a message might have multiple `.` but not be nested (a oneof found inside of a nested message rather than top level). That is what this case handles.

Because all `oneof` field flags are prefixed with the `oneo` name, there will always be a `.`.

Reply:
Oh right, because you're looking for more than one dot. OK.